City-of-Helsinki/drupal-helfi-rekry
# UHF-5296: Search backend

## What has been done
- installed elastic search container
- installed `elasticsearch_connector` and `search_api` modules
- added a search index for job listings with required fields

## How to install
1. get this branch: `git checkout UHF-5296_rekry_search_backend`
2. run `make fresh` to set up the instance
3. run `make drush-cim` to import config

## How to test
- see that the elastic cluster and search index are in place and the correct fields are there;
  - elastic cluster: https://helfi-rekry.docker.so/en/admin/config/search/elasticsearch-connector
  - search API index: https://helfi-rekry.docker.so/en/admin/config/search/search-api
- add the listings to the index at: https://helfi-rekry.docker.so/en/admin/config/search/search-api/index/job_listings (click "Index now") 
- try http://localhost:9200/job_listings/_search to see the search results
  - the port may vary, if 9200 doesn't work, check with `make ps` which port is mapped to that
